pkg/restic/common_test.go

```diff
@@ -0,0 +1,284 @@
+			expected:    map[string]string{"foo": "bar"},
+		},
+		{
+			name:        "has snapshot annotation, with suffix",
+			annotations: map[string]string{"foo": "bar", podAnnotationPrefix + "foo": "bar", podAnnotationPrefix + "abc": "123"},
+			expected:    map[string]string{"foo": "bar", "abc": "123"},
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			pod := &corev1api.Pod{}
+			pod.Annotations = test.annotations
+			assert.EqualValues(t, test.expected, GetPodSnapshotAnnotations(pod))
+		})
+	}
+}
+
+func TestSetPodSnapshotAnnotation(t *testing.T) {
+	tests := []struct {
+		name        string
+		annotations map[string]string
+		volumeName  string
+		snapshotID  string
+		expected    map[string]string
+	}{
+		{
+			name:        "set snapshot annotation on pod with no annotations",
+			annotations: nil,
+			volumeName:  "foo",
+			snapshotID:  "bar",
+			expected:    map[string]string{podAnnotationPrefix + "foo": "bar"},
+		},
+		{
+			name:        "set snapshot annotation on pod with existing annotations",
+			annotations: map[string]string{"existing": "annotation"},
+			volumeName:  "foo",
+			snapshotID:  "bar",
+			expected:    map[string]string{"existing": "annotation", podAnnotationPrefix + "foo": "bar"},
+		},
+		{
+			name:        "snapshot annotation is overwritten if already exists",
+			annotations: map[string]string{podAnnotationPrefix + "foo": "existing"},
+			volumeName:  "foo",
+			snapshotID:  "bar",
+			expected:    map[string]string{podAnnotationPrefix + "foo": "bar"},
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			pod := &corev1api.Pod{}
+			pod.Annotations = test.annotations
+
+			SetPodSnapshotAnnotation(pod, test.volumeName, test.snapshotID)
+			assert.EqualValues(t, test.expected, pod.Annotations)
+		})
+	}
+}
+
+func TestGetVolumesToBackup(t *testing.T) {
+	tests := []struct {
+		name        string
+		annotations map[string]string
+		expected    []string
+	}{
+		{
+			name:        "nil annotations",
+			annotations: nil,
+			expected:    nil,
+		},
+		{
+			name:        "no volumes to backup",
+			annotations: map[string]string{"foo": "bar"},
+			expected:    nil,
+		},
+		{
+			name:        "one volume to backup",
+			annotations: map[string]string{"foo": "bar", volumesToBackupAnnotation: "volume-1"},
+			expected:    []string{"volume-1"},
+		},
+		{
+			name:        "multiple volumes to backup",
+			annotations: map[string]string{"foo": "bar", volumesToBackupAnnotation: "volume-1,volume-2,volume-3"},
+			expected:    []string{"volume-1", "volume-2", "volume-3"},
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			pod := &corev1api.Pod{}
+			pod.Annotations = test.annotations
+
+			assert.EqualValues(t, test.expected, GetVolumesToBackup(pod))
+		})
+	}
+}
+
+func TestGetSnapshotsInBackup(t *testing.T) {
+	tests := []struct {
+		name             string
+		arkBackupName    string
```

Comment:
Since this doesn't appear to vary, should we remove it as a test case variable?

Reply:
done.